### PR TITLE
Remove nyquist limit switch

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request removes the pre-compiler switch to exclude the Nyquist limit calculation and thus partially solves issue #667. 

Not calculating the Nyquist limit makes only sense if computing low frequency radiation - below `omega < pi / (2 * delta_t) ` (factor 2 due to propagation away from the detector at the speed of light). This is already resolved by the grid. In most cases, the Nyquist limit should be taken into account. Thus removing the option avoid misconfiguration by accidentally not activating this limit.

### CODE REVIEW

File: src/picongpu/include/plugins/radiation/Radiation.kernel [hunk 1]
```diff
@@ -51,9 +51,7 @@
 #include "memory/Array.hpp"
 
 
-#if( __NYQUISTCHECK__ == 1 )
 #   include "plugins/radiation/nyquist_low_pass.hpp"
```
Comment (psychocoderHPC) [inline src/picongpu/include/plugins/radiation/Radiation.kernel]: please remove the indention

File: src/picongpu/include/simulation_defines/unitless/radiationConfig.unitless [hunk 2]
```diff
@@ -28,6 +28,7 @@
 
 PMACC_CASSERT_MSG(ENABLE_ELECTRONS_must_set_to_1__change_ENABLE_ELECTRONS_in_file_componentsConfig_param,ENABLE_ELECTRONS==1);
 
+PMACC_CASSERT_MSG( The_Nyquist_limit_needs_to_be_below_one, ( picongpu::radiationNyquist::NyquistFactor < 1.0 ) );
```
Comment (psychocoderHPC) [inline src/picongpu/include/simulation_defines/unitless/radiationConfig.unitless]: Is a negative value allowed? If not please check.

File: src/picongpu/include/plugins/radiation/Radiation.kernel (hunk 1 above)
Comment (PrometheusPi) [inline src/picongpu/include/plugins/radiation/Radiation.kernel]: Good point - will fix it

File: src/picongpu/include/simulation_defines/unitless/radiationConfig.unitless (hunk 2 above)
Comment (PrometheusPi) [inline src/picongpu/include/simulation_defines/unitless/radiationConfig.unitless]: Good idea. The value should be larger than zero. Thus another check would be great.

**Comment by PrometheusPi** [comment]
@psychocoderHPC I updated the pull request.

**Pull request merged**